File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/ConnectionEvent.java
package dev.robocode.tankroyale.botapi.events;

import java.net.URI;

/**
 * Base class of all connection events.
 */
abstract class ConnectionEvent implements IEvent {

    // URI of the server.
    private final URI serverUri;

    /**
     * Initializes a new instance of the ConnectionEvent class.
     *
     * <p>param serverUrl is the URI of the server.
     */
    protected ConnectionEvent(URI serverUri) {
        this.serverUri = serverUri;
    }

    /**
     * Return the URI of the server.
     *
     * @return the URI of the server.
     */
    public URI getServerUri() {
        return serverUri;
    }
}

File: bot-api/dotnet/api/src/events/ConnectionEvent.cs
using System;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Base class of all connection events.
/// </summary>
public abstract class ConnectionEvent : IEvent
{
    /// <summary>URI of the server.</summary>
    public Uri ServerUri { get; }

    /// <summary>
    /// Initializes a new instance of the ConnectionEvent class.
    /// </summary>
    /// <param name="serverUri">URI of the server.</param>
    protected ConnectionEvent(Uri serverUri) => ServerUri = serverUri;
}